### PR TITLE
zh: add the doc about upgrading from 1.0 to 2.0 manually

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What is changed, added or deleted? (Required)

add the doc about upgrading from 1.0 to 2.0 manually

### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v2.0 changes for now)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@toutdesuite @lance6716 PTAL

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+from:
+  host: "172.16.10.81"                           # 对应原 `from.host`
+  port: 3306                                     # 对应原 `from.port`
+  user: "root"                                   # 对应原 `from.user`
+  password: "VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU="   # 对应原 `from.password`
+```
+
+### 数据迁移任务配置文件
+
+对于[数据迁移任务配置文件](task-configuration-file.md)，v2.0.x 基本与 v1.0.x 保持兼容，后续可简单修改后进行使用。
```
Comment (lance6716) [inline zh/dm-upgrade-1.0-2.0-manual.md:97]: prefer to express "no change in current step", "后续可简单修改后进行使用" seems user should change some content in current step

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+## 第 2 步：部署 v2.0.x 集群
+
+使用 TiUP 按所需要节点数部署新的 v2.0.x 集群。
+
+## 第 3 步：下线 v1.0.x 集群
+
+如果原 v1.0.x 集群是使用 DM-Ansible 部署的，则[使用 DM-Ansible 下线 v1.0.x 集群](https://docs.pingcap.com/zh/tidb-data-migration/stable/cluster-operations#%E4%B8%8B%E7%BA%BF%E9%9B%86%E7%BE%A4) 。
```
Comment (lance6716) [inline zh/dm-upgrade-1.0-2.0-manual.md:105]: mention binary deployment

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        > SELECT `id`, `binlog_name`, `binlog_pos` FROM `dm_meta`.`task_v1_syncer_checkpoint` WHERE `is_global`=1;
+        +------------------+-------------------------+------------+
+        | id               | binlog_name             | binlog_pos |
+        +------------------+-------------------------+------------+
+        | mysql-replica-01 | mysql-bin|000001.000123 | 15847      |
+        | mysql-replica-02 | mysql-bin|000001.000456 | 10485      |
+        +------------------+-------------------------+------------+
+        ```
+
+3. 更新 v1.0.x 的数据迁移任务以用于启动新的 v2.0.x 的数据迁移任务。
```
Comment (lance6716) [inline zh/dm-upgrade-1.0-2.0-manual.md:126]:
```suggestion
3. 更新 v1.0.x 的数据迁移任务配置文件以用于启动新的 v2.0.x 的数据迁移任务。
```

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+    - 如 v1.0.x 的数据迁移任务配置文件为 `task_v1.yaml`，则将其复制一份为 `task_v2.yaml`。
+    - 对 `task_v2.yaml` 进行以下修改：
+        - 将 `name` 修改为一个新的、不存在的名称，如 `task_v2`
+        - 将 `task-mode` 修改为 `incremental`
+        - 为各 source 设置在 step.2 中获取的全局 checkpoint 作为增量迁移的起始点，如：
+
+            ```yaml
+            mysql-instances:
+              - source-id: "mysql-replica-01"
```
Comment (lance6716) [inline zh/dm-upgrade-1.0-2.0-manual.md:136]: should we indicate this is `id` from checkpoint? (Of course it's obvious 🤣 )

BTW, seems string in YAML don't need quote

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
+---
+title: TiDB Data Migration 1.0.x 到 2.0.x 手动升级
+summary: 了解 TiDB Data Migration 工具从 1.0.x 到 2.0.2 的手动升级操作。
```
[the remark this answers is not in the document]
Comment (csuzhangxc) [reply zh/dm-upgrade-1.0-2.0-manual.md]: my mistake, fixed in https://github.com/pingcap/docs-dm/pull/189/commits/197a595db1d3db186a48b3320ee6d42a71f65472.

File: zh/dm-upgrade-1.0-2.0-manual.md (hunk 1 above)
Comment (csuzhangxc) [reply zh/dm-upgrade-1.0-2.0-manual.md]: updated in https://github.com/pingcap/docs-dm/pull/189/commits/197a595db1d3db186a48b3320ee6d42a71f65472.

File: zh/dm-upgrade-1.0-2.0-manual.md (hunk 2 above)
Comment (csuzhangxc) [reply zh/dm-upgrade-1.0-2.0-manual.md:105]: mentioned in https://github.com/pingcap/docs-dm/pull/189/commits/197a595db1d3db186a48b3320ee6d42a71f65472.

File: zh/dm-upgrade-1.0-2.0-manual.md (hunk 3 above)
Comment (csuzhangxc) [reply zh/dm-upgrade-1.0-2.0-manual.md]: addressed in https://github.com/pingcap/docs-dm/pull/189/commits/197a595db1d3db186a48b3320ee6d42a71f65472.

File: zh/dm-upgrade-1.0-2.0-manual.md (hunk 4 above)
Comment (csuzhangxc) [reply zh/dm-upgrade-1.0-2.0-manual.md]: add a comment in https://github.com/pingcap/docs-dm/pull/189/commits/197a595db1d3db186a48b3320ee6d42a71f65472.

YES, if no special chars in the string, no quotes needed. but in order to unify the format, I added quotes.

**Comment by lance6716** [comment]
LGTM

**Comment by csuzhangxc** [comment]
@ireneontheway PTAL

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+# TiDB Data Migration 1.0.x 到 2.0.x 手动升级
+
+本文档主要介绍如何手动从 DM v1.0.x 升级到 v2.0.x，主要思路为利用 v1.0.x 时的全局 checkpoint 信息在 v2.0.x 集群中启动一个新的增量数据迁移任务。
+
+> **注意：**
+>
+> - DM 当前不支持在数据迁移任务处于全量导出或全量导入过程中从 v1.0.x 升级到 v2.0.x。
+> - 由于 DM 各组件间用于交互的 gRPC 协议进行了较大变更，因此需确保升级前后 DM 集群各组件（包括 dmctl）使用相同的版本。
+> - 由于 DM 集群的元数据存储（如 checkpoint、shard DDL lock 状态及 online DDL 元信息等）发生了较大变更，升级到 v2.0.x 后无法自动复用 v1.0.x 的元数据，因此在执行升级操作前需要确保：
```
Comment (ireneontheway) [inline zh/dm-upgrade-1.0-2.0-manual.md:14]:
```suggestion
> - 由于 DM 集群的元数据存储（如 checkpoint、shard DDL lock 状态及 online DDL 元信息等）发生了较大变更，升级到 v2.0.x 后无法自动复用 v1.0.x 的元数据，因此在执行升级操作前需要确保所有数据迁移任务不处于：
```

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+>     - 所有数据迁移任务不处于 shard DDL 协调过程中。
```
Comment (ireneontheway) [inline zh/dm-upgrade-1.0-2.0-manual.md:15]: 避免在无序列表每一项开头重复相同的词语。
```suggestion
>     - shard DDL 协调过程中。
```

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+>     - 所有数据迁移任务不处于 online DDL 协调过程中。
```
Comment (ireneontheway) [inline zh/dm-upgrade-1.0-2.0-manual.md:16]:
```suggestion
>     - online DDL 协调过程中。
```

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+## 第 1 步：准备 v2.0.x 的配置文件
+
+### 上游数据库配置文件
+
+在 v2.0.x 中将[上游数据库 source 相关的配置](source-configuration-file.md)从 DM-worker 的进程配置中独立了出来，因此需要根据 [v1.0.x 的 DM-worker 配置](https://docs.pingcap.com/zh/tidb-data-migration/stable/dm-worker-configuration-file) 拆分得到 source 配置。
```
Comment (ireneontheway) [inline zh/dm-upgrade-1.0-2.0-manual.md:22]:
```suggestion
在 v2.0.x 中将[上游数据库 source 相关的配置](source-configuration-file.md)从 DM-worker 的进程配置中独立了出来，因此需要根据 [v1.0.x 的 DM-worker 配置](https://docs.pingcap.com/zh/tidb-data-migration/stable/dm-worker-configuration-file)拆分得到 source 配置。
```

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+
+对于[数据迁移任务配置文件](task-configuration-file.md)，v2.0.x 基本与 v1.0.x 保持兼容，可直接复制 v1.0.x 的配置。
+
+## 第 2 步：部署 v2.0.x 集群
+
+使用 TiUP 按所需要节点数部署新的 v2.0.x 集群。
+
+## 第 3 步：下线 v1.0.x 集群
+
+如果原 v1.0.x 集群是使用 DM-Ansible 部署的，则[使用 DM-Ansible 下线 v1.0.x 集群](https://docs.pingcap.com/zh/tidb-data-migration/stable/cluster-operations#%E4%B8%8B%E7%BA%BF%E9%9B%86%E7%BE%A4) 。
```
Comment (ireneontheway) [inline zh/dm-upgrade-1.0-2.0-manual.md:105]:
```suggestion
如果原 v1.0.x 集群是使用 DM-Ansible 部署的，则[使用 DM-Ansible 下线 v1.0.x 集群](https://docs.pingcap.com/zh/tidb-data-migration/stable/cluster-operations#%E4%B8%8B%E7%BA%BF%E9%9B%86%E7%BE%A4)。
```

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+
+如果原 v1.0.x 集群是使用 Binary 部署，则直接停止 DM-worker 与 DM-master 进程。
+
+## 第 4 步：升级数据迁移任务
+
+1. 使用 [operate-source](manage-source.md#加载数据源配置) 命令将 [准备 v2.0.x 的配置文件](#第-1-步准备-v20x-的配置文件) 中得到的上游数据库 source 配置加载到 v2.0.x 集群中。
+
+2. 在下游 TiDB 中，从 v1.0.x 的数据迁移任务对应的增量 checkpoint 表中获取到对应的全局 checkpoint 信息。
```
Comment (ireneontheway) [inline zh/dm-upgrade-1.0-2.0-manual.md:113]:
```suggestion
2. 在下游 TiDB 中，从 v1.0.x 的数据迁移任务对应的增量 checkpoint 表中获取对应的全局 checkpoint 信息。
```

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+        > SELECT `id`, `binlog_name`, `binlog_pos` FROM `dm_meta`.`task_v1_syncer_checkpoint` WHERE `is_global`=1;
+        +------------------+-------------------------+------------+
+        | id               | binlog_name             | binlog_pos |
+        +------------------+-------------------------+------------+
+        | mysql-replica-01 | mysql-bin|000001.000123 | 15847      |
+        | mysql-replica-02 | mysql-bin|000001.000456 | 10485      |
+        +------------------+-------------------------+------------+
+        ```
+
+3. 更新 v1.0.x 的数据迁移任务配置文件以用于启动新的 v2.0.x 的数据迁移任务。
```
Comment (ireneontheway) [inline zh/dm-upgrade-1.0-2.0-manual.md:128]:
```suggestion
3. 更新 v1.0.x 的数据迁移任务配置文件以启动新的 v2.0.x 数据迁移任务。
```

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+
+    - 如 v1.0.x 的数据迁移任务配置文件为 `task_v1.yaml`，则将其复制一份为 `task_v2.yaml`。
+    - 对 `task_v2.yaml` 进行以下修改：
+        - 将 `name` 修改为一个新的、不存在的名称，如 `task_v2`
```
Comment (ireneontheway) [inline zh/dm-upgrade-1.0-2.0-manual.md:132]:
```suggestion
        - 将 `name` 修改为一个不存在的名称，如 `task_v2`
```

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+        - 将 `task-mode` 修改为 `incremental`
+        - 为各 source 设置在 step.2 中获取的全局 checkpoint 作为增量迁移的起始点，如：
```
Comment (ireneontheway) [inline zh/dm-upgrade-1.0-2.0-manual.md:134]: 这句话可以段句写得更清晰一些吗？

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 15]
```diff
@@ -0,0 +1,155 @@
...
+
+## 第 3 步：下线 v1.0.x 集群
+
+如果原 v1.0.x 集群是使用 DM-Ansible 部署的，则[使用 DM-Ansible 下线 v1.0.x 集群](https://docs.pingcap.com/zh/tidb-data-migration/stable/cluster-operations#%E4%B8%8B%E7%BA%BF%E9%9B%86%E7%BE%A4) 。
+
+如果原 v1.0.x 集群是使用 Binary 部署，则直接停止 DM-worker 与 DM-master 进程。
+
+## 第 4 步：升级数据迁移任务
+
+1. 使用 [operate-source](manage-source.md#加载数据源配置) 命令将 [准备 v2.0.x 的配置文件](#第-1-步准备-v20x-的配置文件) 中得到的上游数据库 source 配置加载到 v2.0.x 集群中。
```
Comment (ireneontheway) [inline zh/dm-upgrade-1.0-2.0-manual.md:111]:
```suggestion
1. 使用 [`operate-source`](manage-source.md#加载数据源配置) 命令将 [准备 v2.0.x 的配置文件](#第-1-步准备-v20x-的配置文件) 中得到的上游数据库 source 配置加载到 v2.0.x 集群中。
```

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 16]
```diff
@@ -0,0 +1,155 @@
+---
```
Comment (yikeke) [inline zh/dm-upgrade-1.0-2.0-manual.md:1]: 建议把文件名改为 `manually-upgrade-dm-1.0-to-2.0.md`，比较容易理解

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 17]
```diff
@@ -0,0 +1,155 @@
...
+title: TiDB Data Migration 1.0.x 到 2.0.x 手动升级
+summary: 了解 TiDB Data Migration 工具从 1.0.x 到 2.0.x 的手动升级操作。
```
Comment (yikeke) [inline zh/dm-upgrade-1.0-2.0-manual.md:3]:
```suggestion
summary: 了解如何从 TiDB Data Migration 1.0.x 手动升级到 2.0.x。
```

File: zh/dm-upgrade-1.0.md [hunk 18]
```diff
@@ -1,11 +1,11 @@
 ---
```
Comment (yikeke) [inline zh/dm-upgrade-1.0.md:1]: 建议把文件名改为 `upgrade-dm-1.0.md`，和 `manually-upgrade-dm-1.0-to-2.0.md` 结构一致

File: zh/dm-upgrade-1.0.md [hunk 19]
```diff
@@ -1,11 +1,11 @@
...
-title: TiDB Data Migration 版本升级
-summary: 了解 TiDB Data Migration 工具的版本变更和升级操作。
+title: TiDB Data Migration 1.0.x 版本间升级
+summary: 了解 TiDB Data Migration 工具的 1.0.x 版本变更和升级操作。
```
Comment (yikeke) [inline zh/dm-upgrade-1.0.md:3]:
```suggestion
summary: 了解 TiDB Data Migration 工具的 1.0.x 版本变更和升级操作。
aliases: ['/zh/tidb-data-migration/dev/dm-upgrade']
```
文件改名需要添加一个 alias，来重定向旧的 url，防止 404。

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 20]
```diff
@@ -0,0 +1,155 @@
...
+> **注意：**
+>
+> - DM 当前不支持在数据迁移任务处于全量导出或全量导入过程中从 v1.0.x 升级到 v2.0.x。
+> - 由于 DM 各组件间用于交互的 gRPC 协议进行了较大变更，因此需确保升级前后 DM 集群各组件（包括 dmctl）使用相同的版本。
+> - 由于 DM 集群的元数据存储（如 checkpoint、shard DDL lock 状态及 online DDL 元信息等）发生了较大变更，升级到 v2.0.x 后无法自动复用 v1.0.x 的元数据，因此在执行升级操作前需要确保：
+>     - 所有数据迁移任务不处于 shard DDL 协调过程中。
+>     - 所有数据迁移任务不处于 online DDL 协调过程中。
+
+## 第 1 步：准备 v2.0.x 的配置文件
+
```
Comment (yikeke) [inline zh/dm-upgrade-1.0-2.0-manual.md:19]:
```suggestion

准备的 v2.0.x 的配置文件包括上游数据库的配置文件以及数据迁移任务的配置文件。

```

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 21]
```diff
@@ -0,0 +1,155 @@
...
+  password: "VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU="   # 对应原 `from.password`
+```
+
+### 数据迁移任务配置文件
+
+对于[数据迁移任务配置文件](task-configuration-file.md)，v2.0.x 基本与 v1.0.x 保持兼容，可直接复制 v1.0.x 的配置。
+
+## 第 2 步：部署 v2.0.x 集群
+
+使用 TiUP 按所需要节点数部署新的 v2.0.x 集群。
```
Comment (yikeke) [inline zh/dm-upgrade-1.0-2.0-manual.md:101]: 这里给个链接？

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 22]
```diff
@@ -0,0 +1,155 @@
...
+
+## 第 3 步：下线 v1.0.x 集群
+
+如果原 v1.0.x 集群是使用 DM-Ansible 部署的，则[使用 DM-Ansible 下线 v1.0.x 集群](https://docs.pingcap.com/zh/tidb-data-migration/stable/cluster-operations#%E4%B8%8B%E7%BA%BF%E9%9B%86%E7%BE%A4) 。
+
+如果原 v1.0.x 集群是使用 Binary 部署，则直接停止 DM-worker 与 DM-master 进程。
```
Comment (yikeke) [inline zh/dm-upgrade-1.0-2.0-manual.md:107]: 「直接停止 DM-worker 与 DM-master 进程」有命令或者运维文档的链接可以提供吗

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 23]
```diff
@@ -0,0 +1,155 @@
...
+                  binlog-pos: 10485
+            ```
+            
+            > **注意：**
+            >
+            > 如在 source 配置中启动了 `enable-gtid`，当前需要通过解析 binlog 或 relay log 文件获取 binlog position 对应的 GTID sets 并在 `meta` 中设置为 `binlog-gtid`。
+
+4. 使用 [`start-task`](create-task.md) 命令以 v2.0.x 的数据迁移任务配置文件启动升级后的数据迁移任务。
+
+5. 使用 [`query-status`](query-status.md) 命令确认数据迁移任务是否运行正常。
```
Comment (yikeke) [inline zh/dm-upgrade-1.0-2.0-manual.md:155]:
```suggestion
5. 使用 [`query-status`](query-status.md) 命令确认数据迁移任务是否运行正常。

如果数据迁移任务运行正常，则证明 DM 升级到 v2.0.x 的操作成功。
```
给个简单的结束语

File: zh/dm-upgrade-1.0-2.0-manual.md [hunk 24]
```diff
@@ -0,0 +1,155 @@
...
+本文档主要介绍如何手动从 DM v1.0.x 升级到 v2.0.x，主要思路为利用 v1.0.x 时的全局 checkpoint 信息在 v2.0.x 集群中启动一个新的增量数据迁移任务。
+
+> **注意：**
+>
+> - DM 当前不支持在数据迁移任务处于全量导出或全量导入过程中从 v1.0.x 升级到 v2.0.x。
+> - 由于 DM 各组件间用于交互的 gRPC 协议进行了较大变更，因此需确保升级前后 DM 集群各组件（包括 dmctl）使用相同的版本。
+> - 由于 DM 集群的元数据存储（如 checkpoint、shard DDL lock 状态及 online DDL 元信息等）发生了较大变更，升级到 v2.0.x 后无法自动复用 v1.0.x 的元数据，因此在执行升级操作前需要确保：
+>     - 所有数据迁移任务不处于 shard DDL 协调过程中。
+>     - 所有数据迁移任务不处于 online DDL 协调过程中。
+
```
Comment (yikeke) [inline zh/dm-upgrade-1.0-2.0-manual.md:17]:
```suggestion

下面是手动升级的具体步骤。

```

File: zh/dm-upgrade-1.0-2.0-manual.md (hunk 21 above)
Comment (csuzhangxc) [reply zh/dm-upgrade-1.0-2.0-manual.md:101]: 有计划给的，不过 TiUP 部署的文档还没有 ready

File: zh/dm-upgrade-1.0-2.0-manual.md (hunk 14 above)
Comment (csuzhangxc) [reply zh/dm-upgrade-1.0-2.0-manual.md]: 在 https://github.com/pingcap/docs-dm/pull/189/commits/da559408ce026459ae041285c3db6e94bb6c340c 更新了一下

File: zh/dm-upgrade-1.0-2.0-manual.md (hunk 16 above)
Comment (csuzhangxc) [reply zh/dm-upgrade-1.0-2.0-manual.md]: 在 https://github.com/pingcap/docs-dm/pull/189/commits/da559408ce026459ae041285c3db6e94bb6c340c 改了

File: zh/dm-upgrade-1.0.md (hunk 18 above)
Comment (csuzhangxc) [reply zh/dm-upgrade-1.0.md]: 在 https://github.com/pingcap/docs-dm/pull/189/commits/da559408ce026459ae041285c3db6e94bb6c340c 改了

File: zh/dm-upgrade-1.0-2.0-manual.md (hunk 22 above)
Comment (csuzhangxc) [reply zh/dm-upgrade-1.0-2.0-manual.md]: 这个没有，用户应该也自己清晰

File: zh/dm-upgrade-1.0-2.0-manual.md (hunk 23 above)
Comment (csuzhangxc) [reply zh/dm-upgrade-1.0-2.0-manual.md]: 在 https://github.com/pingcap/docs-dm/pull/189/commits/cf49ad6437ab1480e1239be149b4d8528d8b37a0 加了

**Comment by csuzhangxc** [comment]
@ireneontheway @yikeke PTAL again

File: zh/dm-upgrade-1.0-2.0-manual.md (hunk 21 above)
Comment (yikeke) [reply zh/dm-upgrade-1.0-2.0-manual.md]: ok，ready 后记得更新

File: zh/dm-upgrade-1.0-2.0-manual.md (hunk 22 above)
Comment (yikeke) [reply zh/dm-upgrade-1.0-2.0-manual.md]: ok

**Pull request merged**